### PR TITLE
Allow CA to be initialised with a crypto.Signer.

### PR DESCRIPTION
**Pull request by kisom**

### CODE REVIEW

**Comment by jsha** [comment]
LGTM, FWIW

**Comment by lziest** [comment]
lgtm

**Comment by kisom** [comment]
Thanks @lziest and @jsha.

File: initca/initca.go [hunk 1]
```diff
@@ -107,10 +108,16 @@ func NewFromPEM(req *csr.CertificateRequest, keyFile string) (cert, csrPEM []byt
...
+	return NewFromSigner(req, priv)
+}
+
+// NewFromSigner creates a new root certificate from a crypto.Signer.
+func NewFromSigner(req *csr.CertificateRequest, priv crypto.Signer) (cert, csrPEM []byte, err error) {
+
 	var sigAlgo x509.SignatureAlgorithm
-	switch priv := priv.(type) {
+	switch pub := priv.Public().(type) {
 	case *rsa.PrivateKey:
```
Comment (jacobhaven) [inline initca/initca.go]: This should probably be `*rsa.PublicKey`

File: initca/initca.go [hunk 2]
```diff
@@ -122,7 +129,7 @@ func NewFromPEM(req *csr.CertificateRequest, keyFile string) (cert, csrPEM []byt
 			sigAlgo = x509.SHA1WithRSA
 		}
 	case *ecdsa.PrivateKey:
```
Comment (jacobhaven) [inline initca/initca.go]: And `*ecdsa.PublicKey`

**Pull request merged**